server/generate.py

```diff
@@ -161,20 +161,60 @@ def gen_enrollment(user, course):
         class_account=gen_maybe(class_account, 0.4),
         section=gen_maybe(section, 0.4))
 
-def gen_backup(user, assignment):
+def gen_messages(assignment, seconds_offset):
+    fizzbuzz_lines = modified_file.split("\n")
+    random.shuffle(fizzbuzz_lines)
+    shuffled_file = "\n".join(fizzbuzz_lines)
+
+    if seconds_offset < 0:
+        progress = abs(seconds_offset)/100100
+    else:
+        progress = (seconds_offset+100000)/100100
+
+    attempts = int(progress * 50)
+    is_passed = True if progress > 0.75 else False
+
+    created = assignment.due_date - datetime.timedelta(seconds=seconds_offset)
+
     messages = {
         'file_contents': {
-            'fizzbuzz.py': modified_file,
+            'fizzbuzz.py': shuffled_file,
             'moby_dick': 'Call me Ishmael.'
         },
-        'analytics': {}
+        'analytics': {
+            'history': {
+                'question': ['fizzbuzz'],
+                'questions': {
+                    'fizzbuzz': {
+                        'attempts': attempts,
+                        'solved': is_passed
+                    }
+                },
+                'all_attempts': attempts
+            },
+            'unlock': gen_bool(),
+            'question': ['fizzbuzz'],
+            'started': {},
```

Comment:
Most of this is cherrypicked from enhancements/schu/diff-view

the change is that I refactored this a bit to separate gen_message from gen_backup